### PR TITLE
[IMP] quality_control_stock: use read_group

### PR DESCRIPTION
**Pull request by kos94ok-3D**

### CODE REVIEW

File: quality_control_stock/models/stock_picking.py [hunk 1]
```diff
@@ -36,6 +24,27 @@ def _compute_count_inspections(self):
     failed_inspections = fields.Integer(
         compute="_compute_count_inspections", string="Inspections failed")
 
+    @api.depends('qc_inspections_ids', 'qc_inspections_ids.state')
+    def _compute_count_inspections(self):
+        Inspection = self.env['qc.inspection']
```
Comment (pedrobaeza) [inline quality_control_stock/models/stock_picking.py]: This is only used once, so no need to let a variable for this.

File: quality_control_stock/models/stock_picking.py [hunk 2]
```diff
@@ -36,6 +24,27 @@ def _compute_count_inspections(self):
...
+        inverse_field = 'picking_id'
```
Comment (pedrobaeza) [inline quality_control_stock/models/stock_picking.py]: Why putting this field name as variable and not `state`. It's used the same number of times... Any way, I prefer in this case to directly use the field name, which is more readable and even shorter in this case.

File: quality_control_stock/models/stock_picking.py [hunk 3]
```diff
@@ -36,6 +24,27 @@ def _compute_count_inspections(self):
...
+        data = Inspection.read_group([
+            ('id', 'in', self.mapped('qc_inspections_ids').ids)
+        ], [inverse_field, 'state'], [inverse_field, 'state'], lazy=False)
+        picking_data = defaultdict(dict)
+        for d in data:
+            picking_data[d[inverse_field][0]].setdefault(d['state'], 0)
+            picking_data[d[inverse_field][0]][d['state']] += d['__count']
+        for picking in self:
+            count_data = picking_data.get(picking.id, {})
+            picking.created_inspections = sum(
```
Comment (pedrobaeza) [inline quality_control_stock/models/stock_picking.py]:
```suggestion
            picking.created_inspections = sum(count_data.values())
```

File: quality_control_stock/models/stock_production_lot.py [hunk 4]
```diff
@@ -32,3 +21,24 @@ def _compute_count_inspections(self):
         compute="_compute_count_inspections", string="Inspections OK")
     failed_inspections = fields.Integer(
         compute="_compute_count_inspections", string="Inspections failed")
+
+    @api.depends('qc_inspections_ids', 'qc_inspections_ids.state')
+    def _compute_count_inspections(self):
+        Inspection = self.env['qc.inspection']
```
Comment (pedrobaeza) [inline quality_control_stock/models/stock_production_lot.py]: Same considerations as in the other.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Good performance improvement!

File: quality_control_stock/models/stock_picking.py (hunk 2 above)
Comment (kos94ok-3D) [reply quality_control_stock/models/stock_picking.py]: Easier copy and past function to another model. If you said that it hardly understand, I`ll fix it.

File: quality_control_stock/models/stock_picking.py (hunk 3 above)
Comment (kos94ok-3D) [reply quality_control_stock/models/stock_picking.py]: Oh, You are right.

**Comment by kos94ok-3D** [comment]
@pedrobaeza Is it ok?

File: quality_control_stock/models/stock_picking.py [hunk 5]
```diff
@@ -36,6 +24,23 @@ def _compute_count_inspections(self):
     failed_inspections = fields.Integer(
         compute="_compute_count_inspections", string="Inspections failed")
 
+    @api.depends('qc_inspections_ids', 'qc_inspections_ids.state')
+    def _compute_count_inspections(self):
+        data = self.env['qc.inspection'].read_group([
+            ('id', 'in', self.mapped('qc_inspections_ids').ids),
+        ], ['picking_id', 'state'], ['picking_id', 'state'], lazy=False)
+        picking_data = defaultdict(dict)
```
Comment (pedrobaeza) [inline quality_control_stock/models/stock_picking.py]: Why do you need `defaultdict`? I think regular dict works as well with `setdefault`

**Comment by pedrobaeza** [comment]
You can include yourself in contributors and copyright in file.

**Comment by LoisRForgeFlow** [comment]
@kos94ok-3D Can you rebase your branch?, travis has been fixed.

**Comment by LoisRForgeFlow** [comment]
@kos94ok-3D have you signed CLA? (https://odoo-community.org/page/cla)

**Comment by kos94ok-3D** [comment]
> @kos94ok-3D have you signed CLA? (https://odoo-community.org/page/cla)

Yes, I have.

**Review by rousseldenis — APPROVED** [review]
LGTM. Code review

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**